src/TopologicalSort.jl

```diff
@@ -0,0 +1,61 @@
+# Compute a topological ordering of a set of nodes reachable from the given
+# roots by the given successor function.
+function topological_sort(successors::F, roots::AbstractVector{Any}) where { F <: Function }
+    # Compute pred_counts, the number of predecessors of each node
+    pred_counts = Dict{Any, Int}()
+    counted = Set{Any}()
+    to_count = Vector{Any}(roots)
+
+    # broken into a separate function to permit specialization on F and N.
+    function count(node::N) where { N }
+        successors(node) do succ
+            push!(to_count, succ)
+            pred_counts[succ] = get(pred_counts, succ, 0) + 1
```

Comment:
```suggestion
            pred_counts[succ] += 1
```
Below code has already guaranteed that `pred_counts` has the key

Reply:
No it hasn't.  The first node to be processed is the root, and when we try to increment the successor's predecessor count, it hasn't been initialized yet.